docs/tools/terraform/howto/upgrade-caching-valkey-terraform.md

```diff
@@ -0,0 +1,78 @@
+---
+title: Update Terraform after Aiven for Caching to Aiven for Valkey™ upgrade
+sidebar_label: Update Terraform configuration after Valkey upgrade
+---
+
+Learn how to update your Terraform configuration after upgrading from Aiven for Caching to Aiven for Valkey™ through the Aiven Console or API.
+
+## Prerequisites
+
+- Verify that you have upgraded your Aiven for Caching service to [Aiven for Valkey through the Aiven Console or API](/docs/products/caching/howto/upgrade-aiven-for-caching-to-valkey).
+- Ensure you have the latest version of [Terraform](/docs/tools/terraform) installed.
+- Review the [Aiven Redis resource documentation](https://registry.terraform.io/providers/aiven/aiven/latest/docs/resources/redis)
+  and the [Aiven Valkey resource documentation](https://registry.terraform.io/providers/aiven/aiven/latest/docs/resources/valkey).
+
+## Step 1: Modify the Terraform Configuration
+
+   - Replace `aiven_redis` and `aiven_redis_user` resources in your configuration
+     with `aiven_valkey` and `aiven_valkey_user`.
+   - Update Aiven for Caching-specific configuration parameters to their
+     Aiven for Valkey equivalents. For example, change `redis_timeout` to `valkey_timeout`.
+
+     **Example configuration:**
+
+     ```hcl
+     # Before: Aiven for Caching configuration
+     resource "aiven_redis" "caching_service" {
+       project      = "PROJECT_NAME"
+       cloud_name   = "CLOUD_REGION"
+       plan         = "PLAN_NAME"
+       service_name = "SERVICE_NAME"
+       redis_user_config {
+         redis_timeout = 300
+       }
+     }
+     ```
+
+     ```hcl
+     # After: Aiven for Valkey configuration
+     resource "aiven_valkey" "valkey_service" {
+       project      = "PROJECT_NAME"
+       cloud_name   = "CLOUD_REGION"
+       plan         = "PLAN_NAME"
+       service_name = "SERVICE_NAME"
+       valkey_user_config {
+         valkey_timeout = 300
+       }
+     }
+     ```
+
+## Step 2: Update the Terraform State
+
+   1. Import each Aiven for Valkey resource into the Terraform state:
+
+      ```bash
+      terraform import aiven_valkey.valkey_service PROJECT_NAME/SERVICE_NAME
+
```

Comment:
We said in step 1 that they should add their users as well, but I'd add an example here to make it clearer:
```
terraform import aiven_valkey_user.user <project-name>/<service-name>/<username>
```

Reply:
Added this as a point. Is this what you mean?

Reply:
yes, looks good